### ISSUE TITLE
node.sh runtime error for an empty multi BLS key folder

### ISSUE THREAD

**Issue by yelllowsin** [harmony-one/harmony#2506]
Current node.sh is not throwing a correct message error when no BLS keys exist on` .hmy/blskeys `folder. Instead it is giving the runtime error bellow. Ideally, the code should be modified to handle this scenario and give the proper error message to the user on screen.

Node Version:

```
./node.sh -V
Harmony (C) 2020. harmony, version v5678-master-20191208.0-301-ga6cf930d (jenkins@ 2020-03-15T22:04:24+0000)
```

```
 ./node.sh -S -N staking -z -M
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   799  100   799    0     0   2428      0 --:--:-- --:--:-- --:--:--  2428
diff: md5sum.txt: No such file or directory
node.sh: downloaded harmony
node.sh: downloaded wallet
node.sh: downloaded bootnode
node.sh: downloaded libmcl.so
node.sh: downloaded libbls384_256.so
node.sh: downloaded libcrypto.so.10
node.sh: public IP address autodetected: 13.85.9.199
node.sh: ############### Running Harmony Process ###############
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xf44969]

goroutine 1 [running]:
main.readMultiBlsKeys.func1(0x1249bd0, 0xc, 0x0, 0x0, 0x15b6260, 0xc0002961e0, 0xc0001fb540, 0xc0001a2a00)
        /mnt/jenkins/workspace/harmony-release/harmony/cmd/harmony/main.go:278 +0x49
path/filepath.Walk(0x1249bd0, 0xc, 0xc000319828, 0xc000319728, 0x95d4b0)
        /usr/local/go/src/path/filepath/path.go:402 +0x6a
main.readMultiBlsKeys(0xc000298180, 0xc0002981a0, 0x518f9a, 0xc00000e060)
        /mnt/jenkins/workspace/harmony-release/harmony/cmd/harmony/main.go:277 +0x10f
main.setupConsensusKey(0x21966e0, 0x1212dc0, 0x43f5ee, 0xc000319aa8)
        /mnt/jenkins/workspace/harmony-release/harmony/cmd/harmony/main.go:346 +0x3f2
main.setupStakingNodeAccount(0x7ffee9d35797, 0x7)
        /mnt/jenkins/workspace/harmony-release/harmony/cmd/harmony/main.go:256 +0x31
main.main()
        /mnt/jenkins/workspace/harmony-release/harmony/cmd/harmony/main.go:645 +0x598
node.sh: node process finished with status 2
node.sh: restarting in 10s...

```

**Comment by JB273** [comment]
Confirmed also on Ubuntu 18.04:
root@OS-Node2:~/harmony/multibls# ./node.sh -N staking -z -M
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   799  100   799    0     0   3566      0 --:--:-- --:--:-- --:--:--  3566
diff: md5sum.txt: No such file or directory
node.sh: downloaded harmony
node.sh: downloaded wallet
node.sh: downloaded bootnode
node.sh: downloaded libmcl.so
node.sh: downloaded libbls384_256.so
node.sh: downloaded libcrypto.so.10
net.core.somaxconn = 1024
net.core.netdev_max_backlog = 65536
net.ipv4.tcp_tw_reuse = 1
net.ipv4.tcp_rmem = 4096 65536 16777216
net.ipv4.tcp_wmem = 4096 65536 16777216
net.ipv4.tcp_mem = 65536 131072 262144
node.sh: public IP address autodetected: 13.94.211.220
Enter passphrase for the BLS key file :
node.sh: ############### Running Harmony Process ###############
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/harmony-one/harmony/internal/configs/node.(*ConfigType).ShardIDFromConsensusKey(0x21b7360, 0x0, 0x0, 0x0)
        /mnt/jenkins/workspace/harmony-release/harmony/internal/configs/node/config.go:286 +0x4a7
main.setupStakingNodeAccount(0x7ffd258037aa, 0x7)
        /mnt/jenkins/workspace/harmony-release/harmony/cmd/harmony/main.go:257 +0x5f
main.main()
        /mnt/jenkins/workspace/harmony-release/harmony/cmd/harmony/main.go:610 +0x598
node.sh: node process finished with status 2
node.sh: restarting in 10s...
^Cnode.sh: received SIGINT
node.sh: terminating update checker (pid 102322)

**Comment by gupadhyaya** [comment]
should be fixed after this PR is merged: https://github.com/harmony-one/harmony/pull/2548 , please recheck and let me know.

**Comment by yelllowsin** [comment]
> should be fixed after this PR is merged: #2548 , please recheck and let me know.

Great, will test it once merged.

**Comment by yelllowsin** [comment]
Thanks, working ok now. Closing this issue.

```
./node.sh -S -N staking -z -M -I
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   799  100   799    0     0   2619      0 --:--:-- --:--:-- --:--:--  2619
node.sh: binaries did not change
node.sh: public IP address autodetected: 13.85.9.199
node.sh: could not find bls key file: ./.hmy/blskeys/*.key
node.sh: exiting
node.sh: terminating update checker (pid 77875)
```
